src/actions/onboardingActions.js

```diff
@@ -226,10 +229,14 @@ export const navigateToAppFlow = (showIncomingReward?: boolean) => {
 
 export const registerWalletAction = (enableBiometrics?: boolean, themeToStore?: string) => {
   return async (dispatch: Dispatch, getState: GetState, api: SDKWrapper) => {
+    const mnemonic = getState()?.wallet?.onboarding?.mnemonic;
+    // in case mnemonic is null (new user) then it will generate new, otherwise will take imported from state
+    const mnemonicPhrase = mnemonic?.original || generateMnemonicPhrase();
+    if (!mnemonic) dispatch(generateWalletMnemonicAction(mnemonicPhrase));
```

Comment:
Indeed it looks weird and I just moved things here because I noticed that we were generating mnemonic on each username check which happens on input change. Other part of whole onboarding refactor will include actions and this piece as well 👍